### PR TITLE
Change return -1 on error, to return a negative on error.

### PR DESCRIPTION
**Pull request by 1bsyl**
Change return -1 on error, to return a negative on error, which allow to be more consistent across functions.

Only comments, are changed. because if the function returns -1, then this is correct.
but application should check error by test "< 0" and not == -1.

eventually, there could be specific/reserved error code for Unsupported(), OutOfMemory(), InvalidParam() ..

### CODE REVIEW

**Comment by 1bsyl** [comment]
Also SDL_RegisterEvents() can returns -1: becuase it's an uint function
```
1105  * \returns the beginning event number, or (Uint32)-1 if there are not enough
1106  *          user-defined events left.
1107  *
1108  * \since This function is available since SDL 3.0.0.
1109  *
1110  * \sa SDL_PushEvent
1111  */
1112 extern DECLSPEC Uint32 SDLCALL SDL_RegisterEvents(int numevents);
```

**Comment by slouken** [comment]
Yes, those should be changed individually as separate PRs. I'm not sure the best way to approach them, but I'm inclined to have them return -1 on error for consistency.

File: include/SDL3/SDL_audio.h [hunk 1]
```diff
@@ -705,7 +705,7 @@ typedef struct SDL_AudioStream SDL_AudioStream;
  * \param dst_format The format of the desired audio output
  * \param dst_channels The number of channels of the desired audio output
  * \param dst_rate The sampling rate of the desired audio output
- * \returns 0 on success, or -1 on error.
+ * \returns 0 on success, or a negative error code.
```
Comment (slouken) [inline include/SDL3/SDL_audio.h:708]: Can we change these to the standard verbiage?
```
 * \returns 0 on success or a negative error code on failure; call
 *          SDL_GetError() for more information.
```

Comment (1bsyl) [reply include/SDL3/SDL_audio.h:708]: yes, I'll try later. do you think if it this is put on 1 line (eg with sed)? this is going to get replaced correctly by the bot ?

Comment (slouken) [reply include/SDL3/SDL_audio.h:708]: Yes, it should be.

**Comment by slouken** [comment]
Can you squash for review?

**Comment by 1bsyl** [comment]
@slouken, done !

File: include/SDL3/SDL_audio.h [hunk 2]
```diff
@@ -748,7 +749,7 @@ extern DECLSPEC int SDLCALL SDL_PutAudioStreamData(SDL_AudioStream *stream, cons
  * \param stream The stream the audio is being requested from
  * \param buf A buffer to fill with audio data
  * \param len The maximum number of bytes to fill
- * \returns the number of bytes read from the stream, or -1 on error
+ * \returns the number of bytes read from the stream, or a negative error code
```
Comment (slouken) [inline include/SDL3/SDL_audio.h:752]: `on failure; call SDL_GetError() for more information.`

File: include/SDL3/SDL_gamepad.h [hunk 3]
```diff
@@ -174,7 +174,7 @@ typedef struct SDL_GamepadBinding
  *
  * \param mappingString the mapping string
  * \returns 1 if a new mapping is added, 0 if an existing mapping is updated,
- *          -1 on error; call SDL_GetError() for more information.
+ *          negative error code; call SDL_GetError() for more information.
```
Comment (slouken) [inline include/SDL3/SDL_gamepad.h:177]: ` or a negative error code on failure; call SDL_GetError() for more information.`

**Comment by slouken** [comment]
Thanks. In the squashed commit there are a bunch of functions which don't have consistent wording. We should make them all consistent, unless the return value has different semantics. This is where having a typedef for the return code would make it obvious when things are consistent.

**Comment by 1bsyl** [comment]
@slouken 
yes, maybe... I can try to see (later) if that possible to automatically update to some SDL_OK / SDL_ERROR / SDL_ErrorCode ?
there will be some issue, maybe that's may be feasible.

**Comment by slouken** [comment]
If we go that route, I would do something like:
```c
typedef int SDL_ReturnCode;
```
and then return that in the API instead of int and update the documentation, for the functions that have that semantic, and leave everything else alone.

**Pull request closed without merging**